### PR TITLE
Relax parser for command line "-filter:.." option

### PR DESCRIPTION
**Pull request by ddur**
For regex-expression filters, parser should accept any character
sequence between < > and between [ ].

### CODE REVIEW

**Comment by molnargab** [comment]
@ddur, could you check the dogfood results of [build 4.6.271](https://ci.appveyor.com/project/sawilde/opencover/build/4.6.271)?
There's a drop in code coverage compared to [4.6.269](https://ci.appveyor.com/project/sawilde/opencover/build/4.6.269).

File: main/OpenCover.Test/Framework/CommandLineParserTests.cs [hunk 1]
```diff
@@ -772,23 +772,79 @@ public void InvalidServiceCommunicationTimeoutThrowsException(string invalidTime
...
+                  "+[]abc*", 
+                  "-<(nunit-agent.*)>[([])]")]
+
+        public void FilterParsing_NonGreedy(string filterArg, string filter0, string filter1, string filter2)
         {
             var parser = new CommandLineParser(GetFilter(filterArg, false).ToArray()).Do(_ => _.ExtractAndValidateArguments());
 
             // assert
             Assert.AreEqual(3, parser.Filters.Count, filterArg);
+            if (parser.Filters.Count == 3) {
```
Comment (molnargab) [inline main/OpenCover.Test/Framework/CommandLineParserTests.cs]: Is this if needed? The test should already have failed if it wouldn't be true.

Comment (ddur) [inline main/OpenCover.Test/Framework/CommandLineParserTests.cs]: If first assertion fails, then 
1) parser.Filters[X] will probably throw exception.
2) additional tests are irrelevant.

Comment (molnargab) [inline main/OpenCover.Test/Framework/CommandLineParserTests.cs]: Unless I'm much mistaken, `Assert.AreEqual` throws an exception if it is not successful. So when the `if` statement is evaluated, it is always true.

Comment (ddur) [inline main/OpenCover.Test/Framework/CommandLineParserTests.cs]: You are probably right, I'm not sure now if further tests will be performed if previous assertion fails. 
Anyway, extra "if" will not harm.

**Comment by ddur** [comment]
@molnargab 
Above that, previous test did not commit any coverage result.
I noticed that this happens from time to time.
See this two identical builds:
https://ci.appveyor.com/project/sawilde/opencover/build/4.6.249
and repeated
https://ci.appveyor.com/project/sawilde/opencover/build/4.6.250

**Comment by ddur** [comment]
@molnargab 
build dogfood-release is running fine on my PC.
Maybe https://github.com/OpenCover/opencover/blob/master/appveyor.yml
should not be ```os: Unstable```

File: main/OpenCover.Test/Framework/CommandLineParserTests.cs (hunk 1 above)
Comment (ddur) [inline main/OpenCover.Test/Framework/CommandLineParserTests.cs]: If you like "if" condition removed, I will not stop you from doing that  :smiley: :+1:

**Comment by molnargab** [comment]
Yeah, actually works on my machine too. 

What's strange is though the first dogfood run hasn't collected any coverage, yet there is no failure message. The second (UITest) then collects the numbers we saw.

**Comment by ddur** [comment]
@molnargab 
Build for pull request before this merge is OK.
[build 270](https://ci.appveyor.com/project/sawilde/opencover/build/4.6.270)

It is same for next build pair 
[build 272](https://ci.appveyor.com/project/sawilde/opencover/build/4.6.272)
[build 273](https://ci.appveyor.com/project/sawilde/opencover/build/4.6.273)

Something is changing after merge. 

Just noticed appweyor link is /projects/sawilde/*
Merge is gone into sawilde branch?

**Comment by ddur** [comment]
@sawilde ?

**Comment by ddur** [comment]
I created new AppVeyor project under my account. Looks fine
https://ci.appveyor.com/project/ddur/opencover-k99vu

**Comment by ddur** [comment]
I was sure few minutes ago that [build 273](https://ci.appveyor.com/project/sawilde/opencover/build/4.6.273) is same as 271 . 
To many similar screens. :see_no_evil:

**Pull request merged**